progress-bar.js

```diff
@@ -0,0 +1,93 @@
+(function(){
+    if (!document.addEventListener ||
+        (  document.documentElement.style.transform === undefined
+        && document.documentElement.style.webkitTransform === undefined)){
+        return;
+    }
+
+    var TRANSFORM = document.documentElement.style.webkitTransform !== undefined ? 'webkitTransform' : 'transform';
+
+    var options = INSTALL_OPTIONS;
+
+    var containerEl = null;
+    var barEl = null;
+    var render = function(){
+        if (!document.body)
+            return;
+
+        if (!containerEl){
+            containerEl = document.createElement('div');
```

Comment:
Keeping a reference to the bar elements allows us to not have to give an id to the element, which is important as in general any given id can be only used on a page once.  So if the user happened to already have an element with id `progress`, the previous method would cause problems.